data/bookkeeping/block_test.go

```diff
@@ -461,3 +461,76 @@ func TestInitialRewardsRateCalculation(t *testing.T) {
 	consensusParams.InitialRewardsRateCalculation = true
 	require.True(t, runTest())
 }
+
+func performRewardsRateCalculation(
+	t *testing.T, consensusParams config.ConsensusParams,
+	rewardsRate uint64,
+	rewardsLevel uint64, rewardsResidue uint64, rewardsRecalculationRound basics.Round,
+	incentivePoolBalance uint64, totalRewardUnits uint64, startingRound basics.Round) bool {
+	require.GreaterOrEqual(t, incentivePoolBalance, consensusParams.MinBalance)
+
+	curRewardsState := RewardsState{
+		RewardsLevel:              rewardsLevel,
+		RewardsResidue:            rewardsResidue,
+		RewardsRecalculationRound: rewardsRecalculationRound,
+		RewardsRate:               rewardsRate,
+	}
+
+	for rnd := int(startingRound); rnd < int(uint64(startingRound)+consensusParams.RewardsRateRefreshInterval+2); rnd++ {
```

Comment:
This many iterations will not necessarily clear out the incentive pool's balance I think. How about running it for `consensusParams.RewardsRateRefreshInterval * 3` iterations?